sdk/src/main/java/com/uid2/sdk/UID2Manager.kt

```diff
@@ -202,17 +254,23 @@ class UID2Manager internal constructor(
         return IdentityPackage(true, PACKAGE_IDENTITY_REFRESHED, identity, REFRESHED)
     }
 
+    /**
+     * The different results from refreshing an identity.
+     */
+    private sealed interface RefreshResult {
+        data class Success(val identity: UID2Identity?, val status: IdentityStatus): RefreshResult
+        data class Failure(val error: Exception): RefreshResult
+    }
+
     /**
      * Refreshes the given Identity.
      */
-    private fun refreshToken(identity: UID2Identity): Job = scope.launch {
+    private suspend fun refreshToken(identity: UID2Identity): Flow<RefreshResult> = flow {
```

Comment:
@mindwalkr-ttd - Following on from your comments in my last PR, i've updated the interface here to return a Flow.